wgpu-hal/src/gles/egl.rs

```diff
@@ -465,10 +340,7 @@ struct Inner {
     version: (i32, i32),
     supports_native_window: bool,
     config: khronos_egl::Config,
-    #[cfg_attr(Emscripten, allow(dead_code))]
-    wl_display: Option<*mut ffi::c_void>,
-    #[cfg_attr(Emscripten, allow(dead_code))]
-    force_gles_minor_version: wgt::Gles3MinorVersion,
+    // XXX: Store RawDisplayHandle for validation in create_surface()?
```

Comment:
Yeah I think we need this. We need to be able to reject when the user has passed a different display handle.

We could also validate this at the wgpu-core layer, which would enforce the correct behavior across all platforms, not just ones where the consequences are bad. I'm leaning towards this solution.

Reply:
Sure, but it's a tad confusing: users might unconditionally pass their `RawDisplayHandle` to instance creation to ensure EGL initialization work, but subsequently use a different `RawDisplayHandle` in Vulkan for that niche use-case I described.

Maybe that's just not something we should support out of the box, or if the user wants it, they should ensure that the instance **is not** initialized with a `RawDisplayHandle`.